### PR TITLE
precision option for formatted strings

### LINKED ISSUE #252: Control over number precision
Hi and thanks for this tool.  It is integrated in another tool that I'm making and is very useful.

My issue (actually a feature request) is control over the number precision.  I have downloaded the latest version 0.6.18.  For my purposes, I've globally replaced ever string that is:

    .15g

with:

    .6g

and like the results better - the amount of precision is great, the files are smaller, and is more easily digestible by LaTeX.  Obviously, a global change like that would upset others, but I'm recommending something simple like a number precision argument (defaults to 15) and would form the number inserted into the aforementioned strings.

I've seen other issues requested like this, but cannot seem to find that issue again right now.

If you like this idea, I would consider making the change myself and submitting it - otherwise, on my local machine I have this change.

By the way, the main impetus for this, which may seem silly, is that I'm developing a tool that works cross-platform and uses matplotlib2tikz.  Each platform comes up with slightly different numbers and it causes trouble for my unit-testing and version control if things keep changing.  I'm thinking with less that full precision, the resulting .tex files shouldn't be changing.

Thanks again

### PR DESCRIPTION
**Pull request by PetePupalaikis**
closes #252

### CODE REVIEW

**Comment by nschloe** [comment]
Instead of specifying a precision, I'd much rather see a `float_format` argument, defaulting to `'%.15e'` or so.

**Comment by PetePupalaikis** [comment]
@nschloe 
My thought process on the 'precision' argument is that it's simply an integer number - and it can be checked as an input argument.  I'm not currently checking the value, but it could be checked for being between 1 and 15 for example and an exception thrown if outside that region.

The format string, in my opinion, offers too much flexibility, certainly more than what is needed.  The goal is simply to shrink the size and complexity (and memory requirements downstream in LaTeX), not necessarily have fine control over the format.

If I modify this to the format string approach, there is a small problem which ought to be addressed which would make the change easier - there are some portions of the code that don't have the '%' character in front of the format string.  I'm not an expert on these things and I'd have to either strip the % from the front of the format string in these places (or maybe leave it? I have no idea).

Anyway, if you insist on the format string change, I will make it.  Do you?  Would you merge the format_string change if made properly?  If I make the change, I'd want it to default to %.15g like your original code.  Is that what you'd prefer? (you said %.15e in your comments)

Also, can you tell me why my tests fail?  Currently, my test for goodness of change was only that I didn't _change_ the already failing pytest results.  Am I doing something wrong in my testing?

**Comment by nschloe** [comment]
The CI fails because the code is not properly formatted. The actual tests don't even run yet. Use
```
make black
```
on the command line.

File: matplotlib2tikz/axes.py [hunk 1]
```diff
@@ -50,9 +50,9 @@ def __init__(self, data, obj):
         # Sort the limits so make sure that the smaller of the two is actually
         # *min.
         xlim = sorted(list(obj.get_xlim()))
-        self.axis_options.append("xmin={:.15g}, xmax={:.15g}".format(*xlim))
+        self.axis_options.append(("xmin={:"+data['float_format'].strip('%')+"}, xmax={:"+data['float_format'].strip('%')+"}").format(*xlim))
```
Comment (nschloe) [inline matplotlib2tikz/axes.py]: `strip`ping is ugly here. Best use `{:15g}` as the default format.

File: matplotlib2tikz/axes.py [hunk 2]
```diff
@@ -365,14 +365,14 @@ def _colorbar(self, colorbar, data):
...
-        mycolormap, is_custom_cmap = _mpl_cmap2pgf_cmap(colorbar.get_cmap())
+        mycolormap, is_custom_cmap = _mpl_cmap2pgf_cmap(colorbar.get_cmap(),data)
         if is_custom_cmap:
             self.axis_options.append("colormap=" + mycolormap)
         else:
             self.axis_options.append("colormap/" + mycolormap)
 
-        self.axis_options.append("point meta min=%.15g" % limits[0])
-        self.axis_options.append("point meta max=%.15g" % limits[1])
+        self.axis_options.append(("point meta min="+data['float_format']+"") % limits[0])
```
Comment (nschloe) [inline matplotlib2tikz/axes.py]: Use `.format()` please.

File: matplotlib2tikz/axes.py [hunk 3]
```diff
@@ -692,14 +692,14 @@ def _handle_linear_segmented_color_map(cmap):
 
     color_changes = []
     for (k, x) in enumerate(X):
-        color_changes.append("rgb(%d%s)=(%.15g,%.15g,%.15g)" % ((x, unit) + colors[k]))
+        color_changes.append(("rgb(%d%s)=("+data['float_format']+","+data['float_format']+","+data['float_format']+")") % ((x, unit) + colors[k]))
```
Comment (nschloe) [inline matplotlib2tikz/axes.py]: Repeating `data["float_format"]` is ugly. Better:
```
",".join(3 * [data["float_format"])
```

File: matplotlib2tikz/save.py [hunk 4]
```diff
@@ -38,6 +38,7 @@ def get_tikz_code(
     extra_tikzpicture_parameters=None,
     dpi=None,
     show_info=True,
+    float_format='%.15g',
```
Comment (nschloe) [inline matplotlib2tikz/save.py]: See above: Use `{:.15g}`.

**Comment by nschloe** [comment]
Closing in favor of #274.

**Pull request closed without merging**